java/src/test/java/com/google/oak/client/OakClientTest.java

```diff
@@ -0,0 +1,136 @@
+//
+
+package com.google.oak.client;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.Test;
+
+public class OakClientTest {
+  /**
+   * This test demonstrates the use of the API provided by the classes and interfaces in {@code
+   * com.google.oak.client}.
+   */
+  @Test
+  public void testSend() {
+    PilotAttestationClient attestationClient = new PilotAttestationClient();
+    PilotOakClient oakClient = new PilotOakClient.Builder(attestationClient).build();
+    Optional<String> response = oakClient.send("Hello!");
+    assertTrue(response.isPresent());
+    assertEquals("Hello!", response.get());
+  }
+
+  private static class PilotOakClient extends OakClient<String, String> {
+    final PilotRpcClient rpcClient;
+    final Encryptor encryptor;
+
+    PilotOakClient(final Builder builder) {
+      byte[] emptyPublicKeyForTest = new byte[0];
+      builder.attestationClient.attest();
+      // In reality implementations of OakClient must handle empty Optionals and unexpected types.
+      rpcClient = (PilotRpcClient) builder.rpcClientProvider.getRpcClient().get();
+      encryptor = builder.encryptorProvider.getEncryptor(emptyPublicKeyForTest).get();
+    }
+
+    @Override
+    public void close() throws Exception {
+      rpcClient.close();
+    }
+
+    @Override
+    Optional<String> send(final String request) {
+      return encryptor.encrypt(request.getBytes())
+          .map(b -> new String(b, StandardCharsets.UTF_8))
+          .flatMap(rpcClient::send)
+          .map(String::getBytes)
+          .flatMap(encryptor::decrypt)
+          .map(b -> new String(b, StandardCharsets.UTF_8));
+    }
+
+    static class Builder extends OakClient.Builder<String, String, Builder> {
+      final PilotAttestationClient attestationClient;
+      Builder(final PilotAttestationClient attestationClient) {
+        super(attestationClient, attestationClient);
+        this.attestationClient = attestationClient;
+      }
+
+      @Override
+      PilotOakClient build() {
+        return new PilotOakClient(this);
+      }
+
+      @Override
+      protected Builder self() {
+        return this;
+      }
+    }
+  }
+
+  private static class PilotAttestationClient
+      implements OakClient.EncryptorProvider, OakClient.RpcClientProvider<String, String> {
+    final AtomicBoolean attested = new AtomicBoolean(false);
+    final PilotRpcClient rpcClient;
+    final Encryptor encryptor = new Encryptor() {
+      @Override
+      public Optional<byte[]> encrypt(final byte[] data) {
+        return Optional.of(data);
+      }
+
+      @Override
+      public Optional<byte[]> decrypt(final byte[] data) {
+        return Optional.of(data);
+      }
+    };
+
+    PilotAttestationClient() {
+      this.rpcClient = new PilotRpcClient();
+    }
+
+    void attest() {
```

Comment:
super nit: I know this is in a test that will probably probalby change in future, but I would prefer something like `set_attested`. `attest` implies providing evidence, but as far as I can tell setting attested to true indicates that the evidence has been verified as well.

Reply:
Thanks. Renamed it to `verifyEvidence` to better demonstrate the expected sequence of events.